src/NUnitFramework/framework/Assert.That.cs

```diff
@@ -50,23 +76,47 @@ static public void That(bool condition, string message, params object[] args)
         /// an <see cref="AssertionException"/>.
         /// </summary>
         /// <param name="condition">The evaluated condition</param>
-        static public void That(bool condition)
+        public static void That(bool condition)
         {
             Assert.That(condition, Is.True, null, null);
         }
 
         #endregion
 
         #region Lambda returning Boolean
+
 #if !NET_2_0
+        
+        /// <summary>
+        /// Asserts that a condition is true. If the condition is false the method throws
+        /// an <see cref="AssertionException"/>.
+        /// </summary> 
+        /// <param name="condition">A lambda that returns a Boolean</param>
+        /// <param name="getExceptionMessage">A function to build the message included with the Exception</param>
+        public static void That(Func<bool> condition, Func<ConstraintResult, string> getExceptionMessage)
```

Comment:
What's the purpose of this overload. That is, will the user be expected to create a Func that takes a ConstraintResult? That's an internal class and is not there for users except for extensibility purposes. Can you show an example of how/why a user would be using this?